### PR TITLE
Refactor LoadCaptureWidget & ConnectToLocalWidget

### PR DESCRIPTION
**Pull request by antonrohr**
The core of this PR is a refactor for the session setup widgets to do exclusivity via QButtonGroup, instead of the custom exclusivity that was done before. 

This is split up into multiple commits because I coded it this way and because I am too lazy to split it up in multiple PRs. All commits work on their own and do have meaningful commit messages.

### CODE REVIEW

File: src/SessionSetup/include/SessionSetup/LoadCaptureWidget.h [hunk 1]
```diff
@@ -15,14 +16,15 @@
...
+#include "SessionSetup/HasRadioButtonInterface.h"
 
 namespace Ui {
 class LoadCaptureWidget;  // IWYU pragma: keep
 }
 
 namespace orbit_session_setup {
 
-class LoadCaptureWidget : public QWidget {
+class LoadCaptureWidget : public QWidget, HasRadioButtonInterface {
```
Comment (beckerhe) [inline src/SessionSetup/include/SessionSetup/LoadCaptureWidget.h:27]: note that `HasRadioButtonInterface` is not a public base class in this case. But I don't think you want private inheritance.

**Review by beckerhe — COMMENTED** [review]
Nice change. I like it a lot. Just one thing: I don't think you need the interface at all to make it work. Is there something that I am missing?

File: src/SessionSetup/include/SessionSetup/LoadCaptureWidget.h (hunk 1 above)
Comment (antonrohr) [reply src/SessionSetup/include/SessionSetup/LoadCaptureWidget.h]: No there is nothing that you are missing. I remove the interface completely

**Review by beckerhe — APPROVED** [review]

**Review by danielfenner — APPROVED** [review]
lgtm!

**Review by danielfenner — APPROVED** [review]

File: src/SessionSetup/ConnectToLocalWidget.cpp [hunk 2]
```diff
@@ -109,4 +91,6 @@ void ConnectToLocalWidget::OnStartOrbitServiceButtonClicked() {
                    });
 }
 
+QRadioButton* ConnectToLocalWidget::GetRadioButton() const { return ui_->radioButton; }
```
[the remark this answers is not in the document]
Comment (antonrohr) [reply src/SessionSetup/ConnectToLocalWidget.cpp:94]: yeah, I just realized that as well

**Pull request merged**